nbconvert/nbconvertapp.py

```diff
@@ -369,18 +385,27 @@ def postprocess_single_notebook(self, write_results):
         if hasattr(self, 'postprocessor') and self.postprocessor:
             self.postprocessor(write_results)
 
-    def convert_single_notebook(self, notebook_filename):
+    def convert_single_notebook(self, notebook_filename, input_buffer=None):
         """Convert a single notebook. Performs the following steps:
 
             1. Initialize notebook resources
             2. Export the notebook to a particular format
             3. Write the exported notebook to file
             4. (Maybe) postprocess the written file
-
+            
+            If input_buffer is not None, convertion is done using buffer as source ignoring
+            the notebook_filename argument.
         """
-        self.log.info("Converting notebook %s to %s", notebook_filename, self.export_format)
-        resources = self.init_single_notebook_resources(notebook_filename)
-        output, resources = self.export_single_notebook(notebook_filename, resources)
+        if not input_buffer:
+            self.log.info("Converting notebook %s to %s", notebook_filename, self.export_format)
+            resources = self.init_single_notebook_resources(notebook_filename)
+            output, resources = self.export_single_notebook(notebook_filename, resources)
+        else:
+            notebook_filename = "notebook"
```

Comment:
This seems unnecessary. You can assume it's a legitimate name even if `input_buffer` is set. Then you can just provide the value `notebook` in line 440.